src/components/NewComment.jsx

```diff
@@ -0,0 +1,87 @@
+import { useState } from "react";
+import { addReviewComment } from "../utils/api";
+import Loading from "./Loading";
+
+const NewComment = ({
+  review,
+  username,
+  setCommentSubmitted,
+  setCommCount,
+  comments,
+  setComments,
+}) => {
+  const [inputComment, setInputComment] = useState("");
+  const [newComment, setNewComment] = useState({});
+  const [commentFailed, setCommentFailed] = useState(false);
+  const [noCommentAdded, setNoCommentAdded] = useState(false);
+
+  const { review_id } = review;
+
+  const handleCommentInput = (e) => {
+    setInputComment(e.target.value);
+  };
+
+  const handleSubmitComment = (e) => {
+    setCommentFailed(false);
+    setCommentSubmitted(false);
+    e.preventDefault();
+    const reqObj = { username: username, body: inputComment };
+
+    if (inputComment.length > 0) {
+      setNoCommentAdded(false);
+
+      setComments((currComments) => [
+        ...comments,
+        {
+          body: inputComment,
```

Comment:
This does not show the author until you refreshed the page, which you might want to change